hw/drivers/sensors/mpu6050/src/mpu6050.c

```diff
@@ -0,0 +1,466 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * resarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <string.h>
+#include <errno.h>
+#include <assert.h>
+
+#include "defs/error.h"
+#include "os/os.h"
+#include "sysinit/sysinit.h"
+#include "hal/hal_i2c.h"
+#include "sensor/sensor.h"
+#include "sensor/accel.h"
+#include "sensor/gyro.h"
+#include "mpu6050/mpu6050.h"
+#include "mpu6050_priv.h"
+
+#if MYNEWT_VAL(MPU6050_LOG)
+#include "log/log.h"
+#endif
+
+#if MYNEWT_VAL(MPU6050_STATS)
+#include "stats/stats.h"
+#endif
+
+#if MYNEWT_VAL(MPU6050_STATS)
+/* Define the stats section and records */
+STATS_SECT_START(mpu6050_stat_section)
+    STATS_SECT_ENTRY(read_errors)
+    STATS_SECT_ENTRY(write_errors)
+STATS_SECT_END
+
+/* Define stat names for querying */
+STATS_NAME_START(mpu6050_stat_section)
+    STATS_NAME(mpu6050_stat_section, read_errors)
+    STATS_NAME(mpu6050_stat_section, write_errors)
+STATS_NAME_END(mpu6050_stat_section)
+
+/* Global variable used to hold stats data */
+STATS_SECT_DECL(mpu6050_stat_section) g_mpu6050stats;
+#endif
+
+#if MYNEWT_VAL(MPU6050_LOG)
+#define LOG_MODULE_MPU6050    (6050)
+#define MPU6050_INFO(...)     LOG_INFO(&_log, LOG_MODULE_MPU6050, __VA_ARGS__)
+#define MPU6050_ERR(...)      LOG_ERROR(&_log, LOG_MODULE_MPU6050, __VA_ARGS__)
+static struct log _log;
+#else
+#define MPU6050_INFO(...)
+#define MPU6050_ERR(...)
+#endif
+
+/* Exports for the sensor API */
+static int mpu6050_sensor_read(struct sensor *, sensor_type_t,
+        sensor_data_func_t, void *, uint32_t);
+static int mpu6050_sensor_get_config(struct sensor *, sensor_type_t,
+        struct sensor_cfg *);
+
+static const struct sensor_driver g_mpu6050_sensor_driver = {
+    mpu6050_sensor_read,
+    mpu6050_sensor_get_config
+};
+
+/**
+ * Writes a single byte to the specified register
+ *
+ * @param The sensor interface
+ * @param The I2C address to use
+ * @param The register address to write to
+ * @param The value to write
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+int
+mpu6050_write8(struct sensor_itf *itf, uint8_t addr, uint8_t reg,
```

Comment:
The device I2C interface address is already present in sensor_itf as "si_addr". It generally gets set in the bsp for onboard sensors and in the creator for offboard ones. The parameter "addr" is not needed. You already set the interface address. All you would have to do here is use the macro "SENSOR_GET_ITF()" to get the interface and then use "si_addr" field.